app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt

```diff
@@ -154,9 +159,114 @@ class LocalSource(
+                                        archiveInputStream.mark(0)
+                                        val comicInfo = xml.decodeFromReader<ComicInfo>(AndroidXmlReader(archiveInputStream, encoding))
+                                        archiveInputStream.reset()
+                                        setAndCopyMetadata(comicInfo, archiveInputStream, manga, folderPath)
+                                    }
+                                    createNoXmlFile = false
+                                    breaker = true
+                                }
+                            }
+                        }
+                        if (breaker) break
+                    } else if (chapter.extension.equals("rar", true) || chapter.extension.equals("cbr", true)) {
+                        withIOContext {
+                            Archive(chapter).use { archive: Archive ->
+                                val comicInfoFileHeader = archive.fileHeaders.firstOrNull { it.fileName == "ComicInfo.xml" }
+
+                                if (comicInfoFileHeader != null) {
+                                    archive.getInputStream(comicInfoFileHeader).buffered().use { archiveInputStream ->
+                                        archiveInputStream.mark(0)
+                                        val comicInfo = xml.decodeFromReader<ComicInfo>(AndroidXmlReader(archiveInputStream, encoding))
+                                        archiveInputStream.reset()
+                                        setAndCopyMetadata(comicInfo, archiveInputStream, manga, folderPath)
+                                    }
+                                    createNoXmlFile = false
+                                    breaker = true
+                                }
+                            }
+                        }
+                        if (breaker) break
+                    }
+                }
+                if (createNoXmlFile) {
+                    File("$folderPath/.noxml").createNewFile()
+                }
+            }
+        } catch (e: Throwable) {
+            logcat(LogPriority.ERROR, e) { "Error setting Metadata from ComicInfo.xml file for ${manga.title}" }
+        }
+
         return manga
     }
 
+    // Calls setMetadataXml() and creates a copy of the ComicInfo.xml from the archive at the chapter root
+    private suspend fun setAndCopyMetadata(comicInfo: ComicInfo, archiveInputStreamOut: InputStream, manga: SManga, folderPath: String?) {
+        if (comicInfo.title != null || comicInfo.writer != null || comicInfo.summary != null ||
+            comicInfo.genre != null || comicInfo.penciller != null
+        ) setMetadataXml(comicInfo, manga)
+
+        val comicInfoFile = File("$folderPath/ComicInfo.xml")
+
+        withIOContext {
+            comicInfoFile.outputStream().use { comicInfoOut ->
+                archiveInputStreamOut.copyTo(comicInfoOut)
+            }
+            archiveInputStreamOut.close()
```

Comment:
`.use` is preferred.
```suggestion
            comicInfoFile.outputStream().use { comicInfoOut ->
                archiveInputStreamOut.use { it.copyTo(comicInfoOut) }
            }
```